packages/shell-api/src/shell-internal-state.ts

```diff
@@ -0,0 +1,238 @@
+      // TODO: warn here
+      return new CursorIterationResult();
+    }
+    return await this.currentCursor._it();
+  }
+  public help(): Help {
+    this.messageBus.emit('mongosh:help');
+    return new Help({
+      help: 'shell-api.help.description',
+      docs: 'https://docs.mongodb.com/manual/reference/method',
+      attr: [
+        {
+          name: 'use',
+          description: 'shell-api.help.help.use'
+        },
+        {
+          name: 'it',
+          description: 'shell-api.help.help.it'
+        },
+        {
+          name: 'show databases',
+          description: 'shell-api.help.help.show-databases'
+        },
+        {
+          name: 'show collections',
+          description: 'shell-api.help.help.show-collections'
+        },
+        {
+          name: '.exit',
+          description: 'shell-api.help.help.exit'
+        }
+      ]
+    });
+  }
+
+  /**
+   * Prepare a `contextObject` as global context and set it as context
+   * Add each attribute to the AsyncWriter also.
+   *
+   * The `contextObject` is prepared so that it can be used as global object
+   * for the repl evaluationi.
+   *
+   * @note The `contextObject` is mutated, it will retain all of its existing
+   * properties but also have the global shell api objects and functions.
+   *
+   * @param {Object} contextObject - contextObject an object used as global context.
+   */
+  setCtx(contextObject: any): void {
+    // Add API methods for VSCode and scripts
+    contextObject.use = this.use.bind(this);
+    contextObject.show = this.show.bind(this);
+    contextObject.it = this.it.bind(this);
+    contextObject.help = this.help.bind(this);
+    contextObject.toIterator = toIterator;
+    contextObject.print = async(arg) => {
+      if (arg.toReplString) {
+        console.log(await arg.toReplString());
+      } else {
+        console.log(arg);
+      }
+    };
+    contextObject.printjson = contextObject.print;
+    Object.assign(contextObject, ShellBson);
+    contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+    contextObject.sh = new Shard(this.currentDb.mongo);
+
+    // Add global shell objects
+    this.asyncWriter.symbols.initializeApiObjects({
+      db: signatures.Database,
+      rs: signatures.ReplicaSet,
+      sh: signatures.Shard,
+      Mongo: signatures.Mongo
+    });
+
+    try {
+      Object.defineProperty(contextObject, 'db', {
+        set: (newDb: any) => {
+          if (newDb.shellApiType && newDb.shellApiType() !== 'Database') {
+            throw new MongoshInvalidInputError('Cannot reassign \'db\' to non-Database type');
+          }
+          this.currentDb = newDb;
+          contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+          contextObject.sh = new Shard(this.currentDb.mongo);
+          this.fetchConnectionInfo();
+          return newDb;
+        },
+        get: () => {
+          return this.currentDb;
+        }
+      });
+    } catch (e) { // java shell, can't use getters/setters
```

Comment:
Oh boy this is unfortunate. :(

Reply:
it's less annoying than it seems, since there are no new `Mongo` objects in java, we don't need to do any of the things that are inside the setter except assign `db`, which we can do here. We also need to set it in `use`, but so be it

Reply:
Glad to hear that